File: app/src/main/java/ch/unstable/ost/test/TestHelper.java
package ch.unstable.ost.test;

import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.NonNull;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import ch.unstable.ost.api.model.ArrivalCheckpoint;
import ch.unstable.ost.api.model.DepartureCheckpoint;
import ch.unstable.ost.api.model.Location;
import ch.unstable.ost.api.model.PassingCheckpoint;
import ch.unstable.ost.api.model.Route;

public class TestHelper {
    public static <T extends Parcelable> T writeAndRead(T parcelable, Parcelable.Creator<T> creator) {
        Parcel parcel = Parcel.obtain();
        parcelable.writeToParcel(parcel, 0);
        parcel.setDataPosition(0);
        T result = creator.createFromParcel(parcel);
        parcel.recycle();
        return result;
    }

    @NonNull
    public static List<PassingCheckpoint> generatePassingCheckpoints(int number) {
        ArrayList<PassingCheckpoint> checkpoints = new ArrayList<>();
        for (int i = 0; i < number; ++i) {
            Date arrival = new Date(number);
            Date departure = new Date(number);
            Location location = generateLocation(number);
            checkpoints.add(new PassingCheckpoint(arrival, departure, location, "" + number));
        }
        return checkpoints;
    }

    @NonNull
    public static Route generateRandomRoute(int number) {
        String shortName = "short " + number;
        String longName = "long" + number;
        List<PassingCheckpoint> stops = generatePassingCheckpoints(25);
        return new Route(shortName, longName, stops);
    }

    @NonNull
    public static DepartureCheckpoint generateDepartureCheckpoint(int number) {
        Date departure = new Date(number);
        String platform = String.valueOf(number);
        Location location = generateLocation(number);
        return new DepartureCheckpoint(departure, platform, location);
    }

    @NonNull
    public static Location generateLocation(int number) {
        return new Location("" + number, Location.StationType.TRAIN, "" + number);
    }

    @NonNull
    public static ArrivalCheckpoint generateArrivalCheckpoint(int number) {
        Date arrival = new Date(number);
        String platform = String.valueOf(number);
        Location location = generateLocation(number);
        return new ArrivalCheckpoint(arrival, platform, location);
    }
}

File: app/src/main/java/ch/unstable/ost/test/TestHelper.kt
package ch.unstable.ost.test

import android.os.Parcel
import android.os.Parcelable

import java.util.Date

import ch.unstable.ost.api.model.ArrivalCheckpoint
import ch.unstable.ost.api.model.DepartureCheckpoint
import ch.unstable.ost.api.model.Location
import ch.unstable.ost.api.model.PassingCheckpoint
import ch.unstable.ost.api.model.Route

import java.util.Arrays.asList

fun generatePassingCheckpoints(number: Int): ArrayList<PassingCheckpoint> {
    val checkpoints: ArrayList<PassingCheckpoint> = ArrayList<PassingCheckpoint>()
    for (i in 0 until number) {
        val arrival = Date(number.toLong())
        val departure = Date(number.toLong())
        val location = generateLocation(number)
        val checkpoint = PassingCheckpoint(arrival, departure, location, "" + number)
        checkpoints.add(checkpoint)
    }
    return checkpoints
}

fun <T : Parcelable> writeAndRead(parcelable: T, creator: Parcelable.Creator<T>): T {
    val parcel = Parcel.obtain()
    parcelable.writeToParcel(parcel, 0)
    parcel.setDataPosition(0)
    val result = creator.createFromParcel(parcel)
    parcel.recycle()
    return result
}

fun generateRandomRoute(number: Int): Route {
    val shortName = "short " + number
    val longName = "long" + number
    val stops = generatePassingCheckpoints(25)
    return Route(shortName, longName, stops)
}

fun generateDepartureCheckpoint(number: Int): DepartureCheckpoint {
    val departure = Date(number.toLong())
    val platform = number.toString()
    val location = generateLocation(number)
    return DepartureCheckpoint(departure, platform, location)
}

fun generateLocation(number: Int): Location {
    return Location("" + number, "" + number, Location.StationType.TRAIN)
}

fun generateArrivalCheckpoint(number: Int): ArrivalCheckpoint {
    val arrival = Date(number.toLong())
    val platform = number.toString()
    val location = generateLocation(number)
    return ArrivalCheckpoint(arrival, platform, location)
}
